libs/pages/login/src/lib/ui/login/login.spec.tsx

```diff
@@ -1,22 +1,174 @@
+  __esModule: true,
+  default: jest.fn(() => ({
+    auth0Error: null,
+    clearError: jest.fn(),
+  })),
+}))
+
+const mockUseAuth0Error = useAuth0Error as jest.MockedFunction<typeof useAuth0Error>
+
+function mockGlobal() {
+  mockUseAuth0Error.mockReturnValue({
+    auth0Error: {
+      error: 'SSO Connection Error',
+      error_description: 'The SSO connection failed. Please check your domain and try again.',
+    },
+    setAuth0Error: jest.fn(),
+  })
+}
+
 describe('Login', () => {
+  const mockOnClickAuthLogin = jest.fn()
   const props: ILoginProps = {
-    onClickAuthLogin: (provider: string) => {
-      return
-    },
+    onClickAuthLogin: mockOnClickAuthLogin,
   }
 
+  beforeEach(() => {
+    jest.clearAllMocks()
+  })
+
   it('should render successfully', () => {
+    mockGlobal()
     const { baseElement } = renderWithProviders(<Login {...props} />)
     expect(baseElement).toBeTruthy()
   })
 
   it('should call invitation detail if token are in the localStorage', async () => {
+    mockGlobal()
     localStorage.setItem('inviteToken', 'token')
     renderWithProviders(<Login {...props} />)
 
     expect(screen.queryByText('Connect to Qovery')).not.toBeInTheDocument()
   })
+
+  it('should display SSO connection error when auth0Error is present', () => {
+    mockGlobal()
+    renderWithProviders(<Login {...props} />)
+
+    // Check that the error message is displayed
+    expect(screen.getByText('SSO Connection Error')).toBeInTheDocument()
+    expect(screen.getByText('The SSO connection failed. Please check your domain and try again.')).toBeInTheDocument()
+  })
+
+  describe('Authentication provider buttons', () => {
+    beforeEach(() => {
+      mockUseAuth0Error.mockReturnValue({
+        auth0Error: null,
+        setAuth0Error: jest.fn(),
+      })
+    })
+
+    it('should call onClickAuthLogin with GITHUB when GitHub button is clicked', () => {
+      renderWithProviders(<Login {...props} />)
+
+      fireEvent.click(screen.getByText('Continue with Github'))
+
+      expect(mockOnClickAuthLogin).toHaveBeenCalledTimes(1)
+      expect(mockOnClickAuthLogin).toHaveBeenCalledWith(AuthEnum.GITHUB)
+    })
+
+    it('should call onClickAuthLogin with GOOGLE_SSO when Google button is clicked', () => {
+      renderWithProviders(<Login {...props} />)
+
+      fireEvent.click(screen.getByText('Continue with Google'))
+
+      expect(mockOnClickAuthLogin).toHaveBeenCalledTimes(1)
+      expect(mockOnClickAuthLogin).toHaveBeenCalledWith(AuthEnum.GOOGLE_SSO)
+    })
+  })
+
+  describe('SSO Form', () => {
+    beforeEach(() => {
+      mockUseAuth0Error.mockReturnValue({
+        auth0Error: null,
+        setAuth0Error: jest.fn(),
+      })
+    })
+
+    it('should go back from SSO form and reset form state', () => {
+      renderWithProviders(<Login {...props} />)
+
+      // First show the SSO form
+      fireEvent.click(screen.getByText('Use Enterprise Single Sign-On'))
```

Comment:
`fireEvent` is an deprectated element you can directly use `useEvent`

Reply:
done